### PR TITLE
@cragispaeth => Drag drop - additional UI tests

### PR DESCRIPTION
**Pull request by eessex**
Had a few more refactors that didn't make the last merge + additional tests.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Running your Dangerfile has Failed
  </td>
    </tr>
  </tbody>
</table>




## Danger has errored

Error: Error

```sh
Error: Cannot find module 'package.json'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at getBabelRC (/home/runner/positron/node_modules/babel-jest/build/index.js:53:41)
    at Object.getCacheKey (/home/runner/positron/node_modules/babel-jest/build/index.js:90:14)
    at ScriptTransformer._getCacheKey (/home/runner/positron/node_modules/jest-runtime/build/ScriptTransformer.js:68:26)
    at ScriptTransformer._getFileCachePath (/home/runner/positron/node_modules/jest-runtime/build/ScriptTransformer.js:93:27)
    at ScriptTransformer.transformSource (/home/runner/positron/node_modules/jest-runtime/build/ScriptTransformer.js:182:32)
    at ScriptTransformer._transformAndBuildScript (/home/runner/positron/node_modules/jest-runtime/build/ScriptTransformer.js:275:40)
```


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by kanaabe** [comment]
Hmm sorry, still working through Danger. Will turn off for now.

File: client/apps/edit/components/section_container/index.coffee [hunk 1]
```diff
@@ -55,18 +55,22 @@ module.exports = React.createClass
...
     dragOverTop = $dragOver.position().top + 20 - window.scrollY
     dragOverCenter = dragOverTop + ($dragOver.height() / 2)
-    mouseBelowCenter = mouseY > dragOverCenter and dragOverID is !@props.sections.length
-    if mouseBelowCenter or dragOverID is @props.dragging + 1
+    mouseBelowCenter = mouseY > dragOverCenter
+    dragOverIsNext = dragOverID is @props.dragging + 1
+    dragOverNotFirst = dragOverID != 0
+    draggingNotLast = @props.dragging != @props.sections.length - 1
+
+    if (dragOverNotFirst and draggingNotLast and mouseBelowCenter) or dragOverIsNext
```
Comment (craigspaeth) [inline client/apps/edit/components/section_container/index.coffee]: Much nicer to read! 👏

File: client/apps/edit/components/section_container/test/index.coffee [hunk 2]
```diff
@@ -47,50 +50,73 @@ describe 'SectionContainer', ->
...
+      @props.draggingHeight = 300
+      rendered = ReactDOMServer.renderToString React.createElement(@SectionContainer, @props)
+      rendered.should.containEql 'edit-section-drag-placeholder" style="height:300px;"'
+
+    it 'does not print a drop zone if dragging over original position', ->
+      @props.dragging = 1
+      rendered = ReactDOMServer.renderToString React.createElement(@SectionContainer, @props)
+      rendered.should.not.containEql 'edit-section-drag-placeholder'
+
+  describe '#getDropZonePosition', ->
```
Comment (craigspaeth) [inline client/apps/edit/components/section_container/test/index.coffee]: 👍  always love me some more test coverage.

**Comment by craigspaeth** [comment]
Well done!

**Pull request merged**